File: model/src/main/java/com/apeiron/igor/model/memory/action/Attack.java
package com.apeiron.igor.model.memory.action;

import com.apeiron.igor.model.db.Person;
import com.apeiron.igor.model.db.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Attack implements Acton {
    @Builder.Default
    private String actionName = "Attack";

    private Integer x;
    private Integer y;
    private Person person;

    private User user;

}

File: web-client/src/main/js/igor-client/src/app/model/game-action/attack.ts
import {Person} from "../person";

export class Attack {
  public actionName: String = "Attack";
  public x: number = null;
  public y: number = null;
  public person:Person = null;

  constructor(x: number, y: number, person: Person) {
    this.x = x;
    this.y = y;
    this.person = person;
  }
}
